### PR TITLE
fix/cip30 addresses as cbor instead of bech32

### PR DESCRIPTION
**Pull request by mirceahasegan**
# Context

Most DAPPs expect and use CBOR addresses.

# Proposed Solution

Return cip30 addresses as Cbor (hex string) instead of Cardano.Address (bech32)

### CODE REVIEW

File: packages/wallet/src/cip30.ts [hunk 1]
```diff
@@ -74,6 +74,15 @@ const compareUtxos = (utxo: Cardano.Utxo, comparedTo: Cardano.Utxo) => {
   return 0;
 };
 
+const cardanoAddressToCbor = (address: Cardano.Address | Cardano.RewardAccount): Cbor =>
+  usingAutoFree((scope) => {
+    const cmlAddr = parseCmlAddress(scope, address.toString());
```
Comment (mirceahasegan) [inline packages/wallet/src/cip30.ts:79]: I am not 100% sure this is the right way to do it for reward accounts too. @mkazlauskas, do you know if this is correct?

File: packages/wallet/src/cip30.ts [hunk 2]
```diff
@@ -172,7 +181,7 @@ export const createWalletApi = (
       if (!rewardAccount) {
         throw new ApiError(APIErrorCode.InternalError, 'could not get reward address');
       } else {
-        return [rewardAccount.toString()];
+        return [cardanoAddressToCbor(rewardAccount)];
```
Comment (mirceahasegan) [inline packages/wallet/src/cip30.ts:184]: Is this a breaking change? Considering the return value was already specified as `Promise<Cbor[]>`?

File: packages/wallet/src/cip30.ts (hunk 1 above)
Comment (mkazlauskas) [reply packages/wallet/src/cip30.ts:79]: Yes this seems to be the correct way to do it for reward accounts too. Would be best to test it e2e with a dapp

File: packages/wallet/src/cip30.ts (hunk 2 above)
Comment (mkazlauskas) [reply packages/wallet/src/cip30.ts:184]: Yes, it should be pointed out in the commit message. It was a bug, wouldn't have happened if Cbor was an OpaqueString :sweat_smile:

File: packages/wallet/src/cip30.ts [hunk 3]
```diff
@@ -74,6 +74,15 @@ const compareUtxos = (utxo: Cardano.Utxo, comparedTo: Cardano.Utxo) => {
   return 0;
 };
 
+const cardanoAddressToCbor = (address: Cardano.Address | Cardano.RewardAccount): Cbor =>
+  usingAutoFree((scope) => {
+    const cmlAddr = parseCmlAddress(scope, address.toString());
+    if (!cmlAddr) {
+      throw new ApiError(APIErrorCode.InternalError, `could transform address ${address} to CBOR`);
```
Comment (rhyslbw) [inline packages/wallet/src/cip30.ts:81]:
```suggestion
      throw new ApiError(APIErrorCode.InternalError, `could not transform address ${address} to CBOR`);
```

File: packages/wallet/src/cip30.ts (hunk 2 above)
Comment (rhyslbw) [reply packages/wallet/src/cip30.ts:184]: Should we add an OpaqueString now then?

**Review by rhyslbw — CHANGES_REQUESTED** [review]

File: packages/wallet/src/cip30.ts (hunk 1 above)
Comment (mirceahasegan) [reply packages/wallet/src/cip30.ts:79]: added e2e test in [cda9ceb](https://github.com/input-output-hk/cardano-js-sdk/pull/599/commits/cda9ceb71622602d026fd3ba637637d419d14d31)

File: packages/wallet/src/cip30.ts (hunk 2 above)
Comment (mirceahasegan) [reply packages/wallet/src/cip30.ts:184]: I will mark it resolved after final squash with commit message updated

**Review by mkazlauskas — APPROVED** [review]
Nice work!

**Review by rhyslbw — DISMISSED** [review]

File: packages/wallet/src/cip30.ts (hunk 2 above)
Comment (rhyslbw) [reply packages/wallet/src/cip30.ts:184]: Should we make a task to convert the alias to an `OpaqueString`?

**Review by rhyslbw — APPROVED** [review]
Nice work @mirceahasegan :rocket:

File: packages/wallet/src/cip30.ts (hunk 2 above)
Comment (mirceahasegan) [reply packages/wallet/src/cip30.ts:184]: > I will mark it resolved after final squash with commit message updated
Updated the commit message, thus marking this as resolved.

**Review by mkazlauskas — APPROVED** [review]

**Pull request merged**